libsession/src/main/java/org/session/libsession/messaging/jobs/AttachmentDownloadJob.kt

```diff
@@ -40,6 +42,31 @@ class AttachmentDownloadJob(val attachmentID: Long, val databaseMessageID: Long)
         // Keys used for database storage
         private val ATTACHMENT_ID_KEY = "attachment_id"
         private val TS_INCOMING_MESSAGE_ID_KEY = "tsIncoming_message_id"
+
+        fun eligibleForDownload(threadID: Long,
+                                storage: StorageProtocol,
+                                messageDataProvider: MessageDataProvider,
+                                databaseMessageID: Long): Boolean {
+            val threadRecipient = storage.getRecipientForThread(threadID)
+            val selfSend = messageDataProvider.isMmsOutgoing(databaseMessageID)
+            val sender = if (selfSend) {
+                storage.getUserPublicKey()
+            } else {
+                messageDataProvider.getIndividualRecipientForMms(databaseMessageID)?.address?.serialize()
+            }
+
+            val contact = sender?.let { storage.getContactWithSessionID(it) }
+            if (threadRecipient == null || sender == null || (contact == null && !selfSend)) {
```

Comment:
1. If they're in a group are they definitely a contact? this part requires that we have a contact for them, even if they're in a group. Is that necessary? (I don't know the full semantic of a `Contact`)

2. I think the last of this is partially built into the next if statement.

```suggestion
            if (threadRecipient == null || sender == null)) {
```

but then you can probably inline the threadRecipient and sender :

```
val threadRecipient = storage.getRecipientForThread(threadID) ?: return false
```

etc.

Reply:
I think it's one of those painful "what do you mean by contact" problem that we have on Android.
From doing some quick tests we were able to see a photo in a group from someone who isn't an actual contact. But this db call probably means something different by "contact", especially with the `isTrusted` property in there.